### PR TITLE
Release automation

### PR DESCRIPTION
**Pull request by Larocceau**
* Add a build script
* Add release action for GH release

### CODE REVIEW

File: Build/Build.fs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+
+Target.create "Publish" (fun _ ->
+    let nugetApiKey = Environment.environVarOrFail "NUGET_API_KEY"
+
+    let nugetArgs =
+        $"""push "{outputFolder}\**\*.nupkg" --api-key {nugetApiKey} --source https://api.nuget.org/v3/index.json"""
+
+    let result =
+        DotNet.exec (fun x -> { x with DotNetCliPath = "dotnet" }) "nuget" nugetArgs
```
Comment (jwthomson) [inline Build/Build.fs]: Can you please change this not to use FAKE?

File: paket.dependencies [hunk 2]
```diff
@@ -2,9 +2,12 @@ source https://api.nuget.org/v3/index.json
 framework: net8.0
 storage: none
 
+nuget Fake.Core.Target
+nuget Fake.DotNet.Paket
 nuget FSharp.Core ~> 8
 
 nuget Fable.Remoting.Giraffe ~> 5
+nuget Fstoolkit.ErrorHandling
```
Comment (jwthomson) [inline paket.dependencies]: Can this be removed?

File: Build/Build.fs (hunk 1 above)
Comment (isaacabraham) [reply Build/Build.fs]: What should it be instead? `System.Diagnostics.Process`?

**Comment by Larocceau** [comment]
@jwthomson this should be ready to merge now. Changes since you last reviewed:
* Replaced using FAKE to "manually" running DotNet
* Set the release version and release notes based on GH-supplied env variables

**Pull request merged**